app/models/account.rb

```diff
@@ -0,0 +1,8 @@
+class Account < ApplicationRecord
+  has_many :account_memberships
+  has_many :users, through: :account_memberships
+
+  delegate :name, to: :profile
+
+  delegated_type :profile, types: %w[Business Developer]
```

Comment:
We're saying here that Developer accounts and Business accounts are both _accounts_ so that they can share things in common and can often be used polymorphically through `Current.account`.

Reply:
Thanks! My understanding of the [original issue](https://github.com/joemasilotti/railsdevs.com/issues/237) is that Accounts represent a "paying entity" e.g. a business that is paying railsdevs. Is my interpretation too narrow? I don't understand why we'd need a "Developer account" I guess 🙏

Reply:
Ah, my bad, after reading the thread more I see that it's also something to hang subscriptions and permissions onto, so they can apply to a group of users, or a single user.

Reply:
My understanding from discussing this with @joemasilotti was that users can have a business profile or a developer profile or both, and in the future we may want users to be able to have more than one business profile.

Additionally, people may want to add more than one user to a single business account. The new architecture covers all of these cases. Users can have many accounts, and each account is either a business or developer profile. They have accounts through account membership, an object that lets us model their role within that account (e.g. owner, manager, employee, etc.)